### PR TITLE
Adds links to sort by: stars, updated_at, forks

### PR DESCRIPTION
**Pull request by asterite**
Now we can see what new projects others are doing :-)

### CODE REVIEW

File: app.cr [hunk 1]
```diff
@@ -9,13 +9,20 @@ def headers
   headers
 end
 
-def crystal_repos
+def crystal_repos(sort)
   client = HTTP::Client.new("api.github.com", 443, true)
-  response = client.get("/search/repositories?q=language:crystal&per_page=100", headers)
+  response = client.get("/search/repositories?q=language:crystal&per_page=100&sort=#{sort}", headers)
```
Comment (zamith) [inline app.cr:18]: I'm a bit concerned about this, since it will make a github API request every time we change the sort order. What do you think of having some sort of local cache? It could be on the server or on the client.

We could also go with this for now, and when/if we have a real problem with, fix it then. Ideas?

File: assets/stylesheets/_colors.sass [hunk 2]
```diff
@@ -9,7 +9,7 @@ $dark-grey: #5B5B5B
 
 $color-main: $crystal-dark
 $color-secondary: $yellowish
-$color-link: $green
+$color-link: $black
```
Comment (zamith) [inline assets/stylesheets/_colors.sass:10]: If we're not using it anymore, we can delete the `$green` variable.

File: assets/stylesheets/_index.sass [hunk 3]
```diff
@@ -37,6 +37,11 @@ a
   color: $text-color-dark
   padding: 50px
 
+  .sorting
```
Comment (zamith) [inline assets/stylesheets/_index.sass:40]: What about this?

``` sass
.sorting
  padding-bottom: 50px
```

It keeps everything aligned on the left, but with a bit of white space between the list and the sorting actions.

File: views/index.ecr [hunk 4]
```diff
@@ -16,10 +16,34 @@
 
     <section class="repos">
       <div class="container">
+        <div class="sorting">
+          Sort by:
+            <% {"stars", "updated", "forks"}.each_with_index do |target_sort, index| %>
```
Comment (zamith) [inline views/index.ecr]: What about moving this to a method in the view model?

**Comment by zamith** [comment]
@asterite Nice to see you around here. :)

File: app.cr (hunk 1 above)
Comment (asterite) [inline app.cr:18]: We can use [redis](https://github.com/stefanwille/crystal-redis), heroku supports it out of the box and it's free. What do you think?

I can do this in a separate pull request, if you want, or you can do it yourself, whichever you prefer.

File: assets/stylesheets/_colors.sass (hunk 2 above)
Comment (asterite) [inline assets/stylesheets/_colors.sass:10]: Sure, go ahead. I didn't realize all variables were used.

File: assets/stylesheets/_index.sass (hunk 3 above)
Comment (asterite) [inline assets/stylesheets/_index.sass:40]: I usually add css so the page doesn't look horrible... I'm a disaster at css, so any alternative you have will surely be better than mine :-)

File: views/index.ecr (hunk 4 above)
Comment (asterite) [inline views/index.ecr]: Sure! In fact we can have the sort options in a constant and use it here and in app.cr where we check the sort param is valid.

**Comment by asterite** [comment]
@zamith :-)

After doing this pull request I realized that crystal might be a good candidate for single-page apps: the view can be done with ecr, the backend in crystal, communication via json, everything typesafe... well, except for the javascript.

File: app.cr (hunk 1 above)
Comment (zamith) [inline app.cr:18]: I'd love the pull request, because I don't have a lot of time right now. But I agree that it can go on a different patch.

File: assets/stylesheets/_index.sass (hunk 3 above)
Comment (zamith) [inline assets/stylesheets/_index.sass:40]: Hahaha. Ok, I'll fix it. (not that this style is great, I might have to make it go through a designer, eventually).

File: views/index.ecr (hunk 4 above)
Comment (zamith) [inline views/index.ecr]: Yes, that would be nice. Can you do it?

**Comment by zamith** [comment]
It could be a good fit for an SPA, but most frameworks handle the templating themselves. Having crystal as an API backend might be a good fit though, because it should respond super fast, especially if it allows using HTTP/2.

As for JS, you have [this](http://www.typescriptlang.org/), which seems to be coming a long nicely (haven't worked with it yet, though).

File: app.cr (hunk 1 above)
Comment (asterite) [inline app.cr:18]: Oh, we can cache it in memory, I guess... redis might be overkill for this.

Comment (zamith) [inline app.cr:18]: That was my first thought, but we can think about it on another pull request. I perfectly fine with that.

**Comment by asterite** [comment]
@zamith I did the fixes and added the cache. I don't know if 30 minutes is OK.

File: models/time_cache.cr [hunk 5]
```diff
@@ -0,0 +1,26 @@
+struct TimeCache(K, V)
```
Comment (zamith) [inline models/time_cache.cr:1]: I like this, but I'm wondering if it belongs here or in Frank. What do you think?

**Comment by zamith** [comment]
@asterite Yes, it should be ok. At least for now.

File: models/time_cache.cr (hunk 5 above)
Comment (asterite) [inline models/time_cache.cr:1]: It could belong there, in the standard library, or a different project, but I don't know if this is general enough.

Comment (zamith) [inline models/time_cache.cr:1]: Makes sense. My only remaining concern with this is that it makes a request per sorting type, instead of only one and doing the sorting on our side.

But thinking it through, I guess it's not a big problem, since we only make the request for the first request in every half an hour. It should be good enough for now.

**Comment by zamith** [comment]
Merged in 0b30b277046a9727c3edeaefd55c662923c4e2e5

File: models/time_cache.cr (hunk 5 above)
Comment (asterite) [inline models/time_cache.cr:1]: Yes, with this we are making at most 144 requests per day, I think GitHub can handle that :-)

About caching just one set of results and doing the sorting, I thought about that, but right now you are only fetching the first 100 repos, and we have more than 100 already (yay!), so for now it makes sense to cache these things separately.

Comment (zamith) [inline models/time_cache.cr:1]: I was thinking of adding a pagination crawler (not a great name), that goes and gets all the repos. What do you think?

Comment (asterite) [inline models/time_cache.cr:1]: Yes, or maybe paginate the results in the site.
